File: projekt3/Sheep.py
from Animal import *

class Sheep(Animal):
    
    __SHEEP_STRENGTH__ = 4
    __SHEEP_INITIATIVE__ = 4
    __COLOR__ = "Gray"
    
    def __init__(self, row, col, world):
        super().__init__(row, col, self.__SHEEP_STRENGTH__, self.__SHEEP_INITIATIVE__, world)
    
    def getName(self):
      return "Sheep"
    
    def getColor(self):
        return self.__COLOR__

    def reproduce(self, row, col):
        self._world_.setOrganism(row, col, Sheep())

File: projekt2/src/proj/java/model/animal/Sheep.java
package proj.java.model.animal;

import java.awt.Color;

import proj.java.model.Animal;
import proj.java.model.World;

public class Sheep extends Animal{

    private static final int SHEEP_STRENGTH = 4;
    private static final int SHEEP_INITIATIVE = 4;

    @Override
    public Color getColor() {
        return Color.LIGHT_GRAY;
    }

    @Override
    public int getInitiative() {
        return SHEEP_INITIATIVE;
    }

    @Override
    public int getStrength() {
        return SHEEP_STRENGTH + getExtraStrength();
    }

    @Override
    protected void reproduce(World world, int row, int col) {
        world.setOrganism(row, col, new Sheep());
    }

}
